spec/object_file_spec.rb

```diff
@@ -3,220 +3,473 @@
 require 'spec_helper'
 
 describe Assembly::ObjectFile do
-  it 'does not run if no input file is passed in' do
```

Comment:
the spec changes are extensive and impossible to compare to the old file, but I am confident I didn't remove any tests.  Here's how I worked:

1.  For specs testing only one method, I put them in a `describe` block for the method.
2.  After 1, for specs testing multiple methods, I created a describe block for each method being tested, and _copied_ the relevant lines to the describe block.   Only then did I remove the line from the spec testing multiple methods.   I did this for each line in the multiple methods until each method called was tested in a describe block.